packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts

```diff
@@ -112,18 +113,32 @@ export default class TableResize implements EditorPlugin {
             this.cacheRects();
         }
 
+        this.isRTL = getComputedStyle(this.editor.getDocument().body, 'direction') == 'rtl';
+
         if (this.tableRectMap) {
             let i = this.tableRectMap.length - 1;
             for (; i >= 0; i--) {
                 const { table, rect } = this.tableRectMap[i];
-                if (
-                    e.pageX >= rect.left - INSERTER_SIDE_LENGTH &&
-                    e.pageX <= rect.right &&
-                    e.pageY >= rect.top - INSERTER_SIDE_LENGTH &&
-                    e.pageY <= rect.bottom
-                ) {
-                    this.setCurrentTable(table, rect);
-                    break;
+                if (this.isRTL) {
+                    if (
+                        e.pageX <= rect.right + INSERTER_SIDE_LENGTH &&
+                        e.pageX >= rect.left &&
+                        e.pageY >= rect.top - INSERTER_SIDE_LENGTH &&
+                        e.pageY <= rect.bottom
+                    ) {
+                        this.setCurrentTable(table, rect);
+                        break;
+                    }
+                } else {
```

Comment:
The "if" block and "else" block has exactly same action, so we should merge them.

You can write some util function to help you get real "right" of rect, e.g.

function getLeftWithDir(rect: Rect, margin: number, isRTL: boolean) {
  return isRTL ? rect.right + margin : rect.left - margin;
}

Then you can call it in a single if block

Reply:
merged